src/home_robot/home_robot/navigation_planner/discrete_planner.py

```diff
@@ -202,27 +202,38 @@ def plan(
         ] = 1
 
         # Check collisions if we have just moved and are uncertain
-        if self.last_action == DiscreteNavigationAction.MOVE_FORWARD:
+        if self.last_action == DiscreteNavigationAction.MOVE_FORWARD or (
+            type(self.last_action) == ContinuousNavigationAction
+            and np.linalg.norm(self.last_action.xyt[:2]) > 0
+        ):
             self._check_collision()
 
-        # High-level goal -> short-term goal
-        # Extracts a local waypoint
-        # Defined by the step size - should be relatively close to the robot
-        (
-            short_term_goal,
-            closest_goal_map,
-            replan,
-            stop,
-            closest_goal_pt,
-            dilated_obstacles,
-        ) = self._get_short_term_goal(
-            obstacle_map,
-            np.copy(goal_map),
-            start,
-            planning_window,
-            plan_to_dilated_goal=use_dilation_for_stg,
-            frontier_map=frontier_map,
-        )
+        try:
+            # High-level goal -> short-term goal
+            # Extracts a local waypoint
+            # Defined by the step size - should be relatively close to the robot
+            (
+                short_term_goal,
+                closest_goal_map,
+                replan,
+                stop,
+                closest_goal_pt,
+                dilated_obstacles,
+            ) = self._get_short_term_goal(
+                obstacle_map,
+                np.copy(goal_map),
+                start,
+                planning_window,
+                plan_to_dilated_goal=use_dilation_for_stg,
+                frontier_map=frontier_map,
+            )
+        except Exception:
```

Comment:
What kinds of exceptions do we see here? Should we print a warning

Reply:
This was added as a precaution because I saw rare crashes. Will add a warning and check if this is hit in the latest episodes.